app/controllers/sessions_controller.rb

```diff
@@ -4,11 +4,23 @@ class SessionsController < ApplicationController
   def create
     auth_hash = request.env['omniauth.auth']
 
-    if auth_hash["uid"]
+    if auth_hash && auth_hash["uid"]
       @user = User.find_or_create_from_omniauth(auth_hash)
```

Comment:
I agree, however in this case the auth data from GitHub is only being used to create their `User` model. The `Student` model that we use for listing the PRs and such is still completely separate and is populated manually from the CSV file as we have done previously.